code/core/write_sdl_ntuple.cc

```diff
@@ -782,6 +843,195 @@ void setGnnNtupleMiniDoublet(SDL::Event* event, unsigned int MD)
+    const float hit4_y = hitsInGPU.ys[hit4];
+    const float hit4_z = hitsInGPU.zs[hit4];
+    const float hit4_r = sqrt(hit4_x * hit4_x + hit4_y * hit4_y);
+    ana.tx->pushbackToBranch<float>("t3_0_r", hit0_r);
+    ana.tx->pushbackToBranch<float>("t3_0_x", hit0_x);
+    ana.tx->pushbackToBranch<float>("t3_0_y", hit0_y);
+    ana.tx->pushbackToBranch<float>("t3_0_z", hit0_z);
+    ana.tx->pushbackToBranch<float>("t3_2_r", hit2_r);
+    ana.tx->pushbackToBranch<float>("t3_2_x", hit2_x);
+    ana.tx->pushbackToBranch<float>("t3_2_y", hit2_y);
+    ana.tx->pushbackToBranch<float>("t3_2_z", hit2_z);
+    ana.tx->pushbackToBranch<float>("t3_4_r", hit4_r);
+    ana.tx->pushbackToBranch<float>("t3_4_x", hit4_x);
+    ana.tx->pushbackToBranch<float>("t3_4_y", hit4_y);
+    ana.tx->pushbackToBranch<float>("t3_4_z", hit4_z);
+
+    /* Sigmas for chi2 calculation 
+     * (stolen from SDL::computeSigmasForRegression and SDL::computeRadiusUsingRegressionk) */
+    std::vector<float> sigmas;
+    float inv1 = 0.01f/0.009f;
+    float inv2 = 0.15f/0.009f;
+    // float inv3 = 2.4f/0.009f; // not used
+    for (auto hit : {hit0, hit2, hit4})
+    {
+        // Get module info
+        unsigned int module = hitsInGPU.moduleIndices[hit];
+        SDL::ModuleType module_type = modulesInGPU.moduleType[module];
+        short module_subdet = modulesInGPU.subdets[module];
+        short module_side = modulesInGPU.sides[module];
+        float module_drdz = modulesInGPU.drdzs[module];
+        float module_slope = modulesInGPU.slopes[module];
+        // Get deltas for sigma calculation
+        float delta1, delta2;
+        bool is_flat;
+        // Category 1: barrel PS flat
+        if (module_subdet == SDL::Barrel and module_type == SDL::PS and module_side == SDL::Center)
+        {
+            delta1 = inv1;//1.1111f;//0.01;
+            delta2 = inv1;//1.1111f;//0.01;
+            module_slope = -999.f;
+            is_flat = true;
+        }
+        // Category 2: barrel 2S
+        else if (module_subdet == SDL::Barrel and module_type == SDL::TwoS)
+        {
+            delta1 = 1.f;//0.009;
+            delta2 = 1.f;//0.009;
+            module_slope = -999.f;
+            is_flat = true;
+        }
+        // Category 3: barrel PS tilted
+        else if (module_subdet == SDL::Barrel and module_type == SDL::PS and module_side != SDL::Center)
+        {
+
+            delta1 = inv1;//1.1111f;//0.01;
+            is_flat = false;
+
+            delta2 = (inv2 * module_drdz/sqrtf(1 + module_drdz * module_drdz));
+        }
+        // Category 4: endcap PS
+        else if (module_subdet == SDL::Endcap and module_type == SDL::PS)
+        {
+            delta1 = inv1;//1.1111f;//0.01;
+            is_flat = false;
+
+            /* Despite the type of the module layer of the lower module index,
+             * all anchor hits are on the pixel side and all non-anchor hits are
+             * on the strip side! */
```

Comment:
Is this comment relevant here? If it is, probably its connections should be made clearer?